app/js/main.js

```diff
@@ -27,19 +27,28 @@ Room.all = function(callback) {
 
 var User = function(data) {
   var data = data || {};
-  this.authenticated = m.prop(data.authenticated || false);
+  this.authenticated = function() {
+    var minutesPassed = new Date(Date.now() - this.createdAt()).getMinutes();
+    if (minutesPassed > 59) {
+      return false;
+    } else {
+      return true;
+    }
+  }
+  this.createdAt = m.prop(data.createdAt || Date.now())
 }
 
-User.prototype.authenticate = function(user) {
+User.prototype.authenticate = function() {
+  var user = new User();
   googleCalendarAPI.checkAuth(function(result) {
     if (result && !result.error) {
-      user.authenticated(true);
+      localStorage.clear();
       localStorage["rooms.access-token"] = JSON.stringify(result);
       User.save(user);
-      m.redraw();
+      m.mount(document.body, MainApp);
```

Comment:
`m.redraw()` or `m.render()` is not enough here. I'm assuming they both use a cached version of `ctrl` when calling the view function of `MainApp`. 

The current solution feels very brute force and there has to be a better way. But it works.